STAR-Delegation/draft-ietf-acme-star-delegation.md

```diff
@@ -603,19 +615,66 @@ certificate has been issued by the CA, the IdO:
 At this point of the protocol flow, the same considerations as in
 {{sec-cname-installation}} apply.
 
+Before forwarding the Order request to the CA, the IdO SHOULD ensure that the
+selected CA supports "unauthenticated GET" by inspecting the relevant settings
+in the CA's `directory` object, as per {{sec-nego-allow-cert-get}}.  If the CA
+does not support "unauthenticated GET" of certificate resources, the IdO MUST
+NOT forward the Order request.  Instead, it MUST move the Order status to
+`invalid` and set the `allow-certificate-get` to `false`.  The same occurs in
+case the Order request is forwarded and the CA does not reflect the
+`allow-certificate-get` setting in its Order resource.  The combination of
+`invalid` status and denied `allow-certificate-get` in the Order resource at
+the IdO provides an unambiguous (asynchronous) signal to the NDC about the
+failure reason.
+
 ### Capability Discovery
 
 In order to help a client to discover support for this profile, the directory
-object of an ACME server contains the following attribute in the `meta`
-field:
+object of an ACME server (typically, one deployed by the IdO) contains the
+following attribute in the `meta` field:
 
-- delegation-enabled: boolean flag indicating support for the profile
-  specified in this memo.  An ACME server that supports this delegation profile
-  MUST include this key, and MUST set it to true.
+- delegation-enabled (optional, boolean): Boolean flag indicating support for
+  the profile specified in this memo.  An ACME server that supports this
+  delegation profile MUST include this key, and MUST set it to true.
 
 The `delegation-enabled` flag may be specified regardless of the existence or
 setting of the `auto-renewal` flag.
 
+In order to help a client to discover support for certificate fetching using
+unauthenticated HTTP GET, the directory object of an ACME server (typically,
+one deployed by the CA) contains the following attribute in the `meta` field:
+
+- allow-certificate-get (optional, boolean): See {{sec-nego-allow-cert-get}}.
+
+### Negotiating an Unauthenticated GET
+{: #sec-nego-allow-cert-get}
+
+In order to enable the name delegation of non-STAR certificates, this document
+defines a mechanism that allows a server to advertise support for accessing
+certificate resources via unauthenticated GET (in addition to
+POST-as-GET), and a client to enable this service with per-Order granularity.
+
```

Comment:
I think the rest of this subsection is confusing in duplicating Sec. 3.4 of RFC 8739. Instead we could say:
> A CA that wishes to vend delegated certificates (both STAR and non-STAR) MUST support the `allow-certificate-get` metadata flag, with the same syntax and semantics as in Sec. 3.4 of RFC 8739. Similarly, it MUST support `newOrder` requests containing an `allow-certificate-get` attribute.

Reply:
I think we need to restate the rules because the protocol elements we are talking about are different (same names, but different nesting level)

Reply:
They are both different and similar, which is doubly confusing. The metadata element is unchanged, but the order element is subtly different.

Let's just make sure that someone who does read RFC 8739 still understands how they are related.

Reply:
> The metadata element is unchanged, but the order element is subtly different.

mmm, both are subtly different.  Unless I'm confused -- which might well be!

Reply:
Yes - see my comment below.